src/IK/include/BipedalLocomotion/IK/Task.h

```diff
@@ -0,0 +1,117 @@
+/**
+ * @file Task.h
+ * @authors Giulio Romualdi
+ * @copyright 2021 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_IK_TASK_H
+#define BIPEDAL_LOCOMOTION_IK_TASK_H
+
+#include <memory>
+
+#include <Eigen/Dense>
+
+#include <iDynTree/Core/MatrixView.h>
+#include <iDynTree/KinDynComputations.h>
+
+#include <BipedalLocomotion/ParametersHandler/IParametersHandler.h>
+#include <BipedalLocomotion/System/VariablesHandler.h>
+
+namespace BipedalLocomotion
+{
+namespace IK
+{
+
+/**
+ * Task describes a IK task element. The element is described by a matrix
+ * \f$A\f$ and a vector \f$b\f$. This class describes both a linear equality constraint and a linear
+ * inequality constraint. In case of equality constraint \f$ A \f$ and \f$ b \f$ represents: \f$ Ax
+ * = b\f$ In case of inequality constraint \f$ A \f$ and \f$ b \f$ represents: \f$ Ax \le b \f$
+ * @note Please inherit this class if you want to build your own optimal control problem.
+ */
+class Task
```

Comment:
As discussed F2F, the name ``Task`` is very generic, but the interface is suitable only for linear tasks.

Reply:
What do you think of `LinearTask`?

Reply:
Sounds good!

Reply:
Here the commits 7975cd4e22e520aa719241b8bee4e55b8b4acd94 8d17d8c2aa86762f8dc8f4af3233fbba44cd0af7 91bea8d9233cb835fd296731156de5aa87009364 eebe87c2d2352d10121c4bdca4da810cb728acc5 8943124cb5a410c3998782790bc3ca5c030886c8 cc617c066233c0f549dc886d2e898e65cc80d081 51f396a1048beccd68713051cc22eea48f8cdfa9 d57d384474d63dc70535882d2ba72490c4831a54